src/go/k8s/pkg/resources/certmanager/pki.go

```diff
@@ -119,3 +118,8 @@ func (r *PkiReconciler) Ensure(ctx context.Context) error {
 func (r *PkiReconciler) issuerNamespacedName(name string) types.NamespacedName {
 	return types.NamespacedName{Name: r.pandaCluster.Name + "-" + name, Namespace: r.pandaCluster.Namespace}
 }
+
+// CertificateNames provides names of all certificates for redpanda listeners
+func (r *PkiReconciler) CertificateNames() *resources.Certificates {
+	return resources.NewCertificates(r.pandaCluster)
+}
```

Comment:
Can this creation (new allocation) be done in the `NewPki` and stored in the struct `PkiReconciler`? This function is called multiple time, so it generates additional waste in therms of duplicated instance of `resources.Certificates`.

Reply:
good point, will do that.